File: 530. Minimum Absolute Difference in BST/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function(root) {
  let prev = null;
  let min = Number.MAX_SAFE_INTEGER;

  const traversal = root => {
    if (!root) return;
    traversal(root.left);
    if (prev !== null && min > Math.abs(root.val - prev)) {
      min = Math.abs(root.val - prev);
    }
    prev = root.val;
    traversal(root.right);
  };
  traversal(root);
  return min;
};

File: 530. Minimum Absolute Difference in BST/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
import "math"

func getMinimumDifference(root *TreeNode) int {
	prev := -1
	min := math.MaxInt64
	var dfs func(root *TreeNode)
	dfs = func(root *TreeNode) {
		if root != nil {
			dfs(root.Left)
			if prev != -1 && min > root.Val-prev {
				min = root.Val - prev
			}
			prev = root.Val
			dfs(root.Right)
		}
	}

	dfs(root)
	return min
}